packages/client/src/pages/LeaderboardPage/LeaderboardPage.tsx

```diff
@@ -4,43 +4,52 @@ import { leaderboardFields } from './data';
 import { LeaderboardField } from './LeaderboardField';
 import { LeaderboardRow } from './LeaderboardRow';
 import styles from './leaderboard.module.scss';
-import { LeaderboardData } from './types';
-
-const MOCK_DATA: LeaderboardData[] = [
-  {
-    avatar: null,
-    name: 'ivanivi',
-    score: 16789988,
-  },
-  {
-    avatar: null,
-    name: 'petrovi',
-    score: 15000,
-  },
-  {
-    avatar: null,
-    name: 'mishapetrov',
-    score: 14899,
-  },
-];
+import { useEffect, useState } from 'react';
+import { useGetLeaderboardMutation } from 'api';
+import { GetLeaderboardType, LeaderboardData } from './types';
 
 function LeaderboardPage(): JSX.Element {
+  const [getLeaderboard] = useGetLeaderboardMutation();
+  const [leaderbordList, setLeaderboardList] = useState<LeaderboardData[]>([]);
+
+  const submitData: GetLeaderboardType = {
+    ratingFieldName: 'points',
+    cursor: 0,
+    limit: 10,
+  };
+
+  const getLeaderboardList = async (data: GetLeaderboardType) => {
+    try {
+      const list = await getLeaderboard(data).unwrap();
+      setLeaderboardList(list);
+    } catch (err) {
+      throw new Error((err as Error).message);
+    }
+  };
+
+  useEffect(() => {
+    getLeaderboardList(submitData);
+  }, []);
+
   return (
-    <section className={styles.wrapper}>
-      <table className={styles.leaderboard}>
-        <thead>
-          <tr>
-            {leaderboardFields.map(field => (
-              <LeaderboardField key={field.id} {...field} />
-            ))}
-          </tr>
-        </thead>
-        <tbody>
-          {MOCK_DATA.map((data, index) => (
-            <LeaderboardRow key={index} {...data} place={index + 1} />
-          ))}
-        </tbody>
-      </table>
+    <section className={styles.leaderboard}>
+      <ul className={styles.leaderboardTitles}>
+        {leaderboardFields.map(field => (
+          <LeaderboardField key={field.id} {...field} />
+        ))}
+      </ul>
+      <ul className={styles.list}>
+        {leaderbordList.map((data, index) => (
+          <LeaderboardRow
+            key={data.data.userId}

```

Comment:
Можно в leaderbordList класть данные более плоской структуры, чтобы не приходилось писать data.data....

Reply:
Как я понимаю, не получится, т.к. я отправляю на бекенд объект data и он мне возвращает в объекте data просто весь мой объект data, поэтому такая вложенность получается

Reply:
Тогда можно хотя бы так сделать:
```suggestion
            leaderbordList.map(({ data }, index) => (
                <LeaderboardRow
                      key={data.userId}
                      ...

```

Или даже так:
```suggestion
            leaderbordList.map(
          (
            { data: { userId, points, userAvatar, userNickname, time } },
            index
          ) => (
            <LeaderboardRow
              key={userId}
              points={points}
              userAvatar={userAvatar}
              userNickname={userNickname}
              time={time}
              place={index + 1}
            />
          )
        )

```